pkgs/stdenv/generic/check-meta.nix

```diff
@@ -304,6 +304,11 @@ let
     outputsToInstall = listOf str;
     position = str;
     available = unspecified;
+    availability = unspecified;
```

Comment:
> Because then I don't think we really need to add this to this list

I tried removing this and the meta check failed.

Ideally I'd like to give types for the subcomponents of `availability` (see the following four lines which are commented out) -- but apparently `lib.types` doesn't have "struct types" like yants does?  Or maybe I just can't figure out how to express them.

> At the very least, this should not be located in the "Weirder stuff that doesn't appear in the documentation?" section

If this PR is merged **I will absolutely update the documentation.**

I'd like to do that in a separate PR because our documentation standards are much higher now (hooray!) and mixing together feedback on the code vs feedback on the documentation tends to make both discussions harder to follow.

Reply:
Yeah I've been struggling with this a lot too when trying to implement RFC 127. `lib.types` has clumsy struct types in form of submodules, however checking them is a pain (it `throw`s, and we have to `tryEval` to get around that, at the cost of losing any helpful error messages). See https://github.com/NixOS/nixpkgs/pull/177272#issuecomment-1621946623 for more details.

The solution I'm tentatively going for is https://github.com/NixOS/nixpkgs/pull/177272/files#diff-8d6de52a69f8de7972d373a09d7a0747f88017b71242cd7bc241d61dd431101dR468-R537, I could factor out the changes to `checkMetaAttr` if you want to make use of that as well.

Reply:
> For now, maybe just move those lines into a section commented with "automatically generated in check-meta" or something

Implemented in 4a593cdd0542893b16934cc7553c1cde95b6a792